cmd/snap/cmd_run.go

```diff
@@ -1462,11 +1462,21 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, runner runnable, beforeExec fun
 			if err != cgroup.ErrCannotTrackProcess {
 				return err
 			}
-			// If we cannot track the process then log a debug message.
-			// TODO: if we could, create a warning. Currently this is not possible
-			// because only snapd can create warnings, internally.
-			logger.Debugf("snapd cannot track the started application")
-			logger.Debugf("snap refreshes will not be postponed by this process")
+			switch runner.info.Base {
+			case "", "core", "core18", "core20", "core22":
+				// If we cannot track the process then log a debug message.
+				// TODO: if we could, create a warning. Currently this is not possible
+				// because only snapd can create warnings, internally.
+				logger.Debugf("snapd cannot track the started application")
+				logger.Debugf("snap refreshes will not be postponed by this process")
+			default:
+				// For apps using core24+, fail hard.
```

Comment:
Could you please a comments on why we need to fail hard for core24+?

Reply:
We needed a cut-off point and I _think_ that we selected core24 as the one where it ought to fail. CC @pedronis as IIRC it came from you in an informal context.